source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACROverflowTarget.mm

```diff
@@ -0,0 +1,169 @@
+//
+//  ACROverflowTarget
+//  ACROverflowTarget.mm
+//
+//  Copyright © 2021 Microsoft. All rights reserved.
+//
+
+#import "ACROverflowTarget.h"
+#import "ACOActionOverflow.h"
+#import "ACOBaseActionElementPrivate.h"
+#import "ACRShowCardTarget.h"
+#import "ACRViewPrivate.h"
+#import "UtiliOS.h"
+
+@implementation ACROverflowMenuItem {
+    std::shared_ptr<AdaptiveCards::BaseActionElement> _action;
+    NSObject<ACRSelectActionDelegate> *_target;
+}
+
++ (instancetype)initWithActionElement:(ACOBaseActionElement *)actionElement
+                               target:(NSObject<ACRSelectActionDelegate> *)target
+{
+    ACROverflowMenuItem *item = [[ACROverflowMenuItem alloc] init];
+    item->_action = actionElement.element;
+    item->_target = target;
+    return item;
+}
+
+- (NSString *)title
+{
+    return [NSString stringWithUTF8String:_action->GetTitle().c_str()];
+}
+
+- (NSString *)iconUrl
+{
+    return [NSString stringWithUTF8String:_action->GetIconUrl().c_str()];
+}
+
+- (NSObject<ACRSelectActionDelegate> *)target
+{
+    return _target;
+}
+
+- (UIImage *)iconImageWithSize:(CGSize)size
+{
+    auto &iconUrl = _action->GetIconUrl();
+    if (!iconUrl.empty()) {
+        NSString *url = [NSString stringWithUTF8String:iconUrl.c_str()];
+        UIImage *image = [UIImage imageWithData:[NSData dataWithContentsOfURL:[NSURL URLWithString:url]]];
```

Comment:
Overall, the code is very neat. the only remaining concern I have is this line for synchronous image loading
ACRView has loadImagesForActionsAndCheckIfAllActionsHaveIconImages and - (void)observeValueForKeyPath:(NSString *)path ofObject:(id)object change:(NSDictionary *)change context:(void *)context for getting images via KVO. 
The KVO's key is "image" in ImageView, and it observers for the changes in "image". so you may have to tweak the codes. If you believe that this is the outside of scope for the task, Please create an issue to track it.

Reply:
Revised the logic to follow what we've already done for ACRButton where the icon images are already asynchronously loaded & cached in rootView's image map and in here we just follow the same logic done in ACRButton to load it out. No more sync loading & in-place loading.